### PR TITLE
Allow Users to Set Tale Licenses

### PR DESCRIPTION
**Pull request by ThomasThelen**
### Tale Licenses

This PR adds a feature that allows users to select licenses for their Tales. 

Highlights:
   1. User can select the license in the metadata tab under Run
   2. Tales default to CC-BY (see branch in girder_wholetale)
   3. Dropdown automatically selects current Tale license

#### Testing
   1. Check this branch out
   2. Check out tale_license in girder_wholetale
   3. Deploy
   4. Create a new Tale
   5. Verify it has CC-BY via Girder API or metadata tab
   6. Navigate to Run > metadata
   7. Confirm CC-BY is selected
   8. Change the license and save
   9. Confirm new license state via Girder API or refresh the page & make sure it is selected

### CODE REVIEW

File: app/components/ui/tale-tab-metadata/component.js [hunk 1]
```diff
@@ -27,19 +28,37 @@ export default Component.extend({
...
   
   selectDefaultImageId() {
     // Select the current imageId by default
     const selectedImageId = this.get('model').get('tale').get('imageId');
     $('.ui.dropdown').dropdown('set selected', selectedImageId);
   },
+
+  selectDefaultLicense() {
+    // Select the license that the Tale currently has
+    const selectedLicense = this.get('model').get('tale').get('license');
```
Comment (craig-willis) [inline app/components/ui/tale-tab-metadata/component.js]: Based on merged girder_wholetale PR, this should be `licenseSPDX`

File: app/models/tale.js [hunk 2]
```diff
@@ -18,5 +18,6 @@ export default DS.Model.extend({
   "title": DS.attr('string'),
   "updated": DS.attr('date'),
   "dataSet": DS.attr(),
-  "folderId": DS.attr('string')
+  "folderId": DS.attr('string'),
+  "license": DS.attr('string')
```
Comment (craig-willis) [inline app/models/tale.js]: Based on merged girder_wholetale PR, this should be `licenseSPDX`

File: app/components/ui/tale-tab-metadata/component.js [hunk 3]
```diff
@@ -60,5 +79,10 @@ export default Component.extend({
       const tale = this.get('model').get('tale');
       tale.set('imageId', selected);
     },
+
+    setTaleLicense: function(selected) {
+      const tale = this.get('model').get('tale');
+      tale.set('license', selected);
```
Comment (craig-willis) [inline app/components/ui/tale-tab-metadata/component.js]: Based on merged girder_wholetale PR, this should be licenseSPDX

**Comment by ThomasThelen** [comment]
Good catch @craig-willis - I forgot to update those to reflect the girder_wholetale changes. This should now be resolved.

File: app/components/ui/tale-tab-metadata/component.js [hunk 4]
```diff
@@ -27,19 +28,37 @@ export default Component.extend({
       component.set('environments', images);
       component.selectDefaultImageId();
     });
+
+    // Fetch licenses for users to select
+    $.getJSON(this.get('apiHost') + '/api/v1/license/').then(function(licenses) {
+      component.set('licenses', licenses);
+      component.selectDefaultLicense();
+    });
```
Comment (bodom0015) [inline app/components/ui/tale-tab-metadata/component.js]: Missing error-handling here, but probably not necessary in this case (unless `GET /license` is a more complex endpoint than it seems)

File: app/components/ui/tale-tab-metadata/template.hbs [hunk 5]
```diff
@@ -38,6 +38,24 @@
         </div>
     </div>
 
+    <div class="inline fields ui grid">
+        <label class="two wide column right aligned">License</label>
+        <div class="field thirteen wide column">
+            <div id="licenseDropdown" class="ui icon selection dropdown license {{if (eq licenses.length 0) "loading"}}">
```
Comment (bodom0015) [inline app/components/ui/tale-tab-metadata/template.hbs]: Adding the following to `main.scss` should make the width of this field the same as the others on the form:
```css
#licenseDropdown {
    width: 100%;
}
```

File: app/components/ui/tale-tab-metadata/component.js (hunk 4 above)
Comment (craig-willis) [reply app/components/ui/tale-tab-metadata/component.js]: @ThomasThelen please address this question.  Did you choose jQuery for a reason?

File: app/components/ui/tale-tab-metadata/template.hbs (hunk 5 above)
Comment (craig-willis) [reply app/components/ui/tale-tab-metadata/template.hbs]: @ThomasThelen  please make this change before merging

File: app/components/ui/tale-tab-metadata/component.js (hunk 4 above)
Comment (ThomasThelen) [reply app/components/ui/tale-tab-metadata/component.js]: I think I chose it based on the call to the images a few lines higher in the source; I'll make a query to the store though to keep things as Ember as possible

**Pull request merged**